libr/parse/p/parse_x86_pseudo.c

```diff
@@ -485,6 +485,13 @@ static bool varsub (RParse *p, RAnalFunction *f, ut64 addr, int oplen, char *dat
 		}
 	}
 
+    // Strip brackets from lea pseudo
+	if (r_regex_match ("[a-z]{3} = \\[[a-z0-9_]+\\]$", "ne", tstr) &&
```

Comment:
There's no need to use a regex to achieve this, actually this was the actual behaviour.

See 17b17d91c2926e2900bfa4b34c83b88aa69bcc4b

it just a matter of reverting 06311f718b16176e8c1bbcf4ca36da23286c60a0

Reply:
Also, please add a test for this

Reply:
> There's no need to use a regex to achieve this, actually this was the actual behaviour.
> 
> See [17b17d9](https://github.com/radareorg/radare2/commit/17b17d91c2926e2900bfa4b34c83b88aa69bcc4b)
> 
> it just a matter of reverting [06311f7](https://github.com/radareorg/radare2/commit/06311f718b16176e8c1bbcf4ca36da23286c60a0)

Not quite sure what you mean, because simply reverting [06311f7](https://github.com/radareorg/radare2/commit/06311f718b16176e8c1bbcf4ca36da23286c60a0) indeed strips brackets, but at the same time breaks variable name resolution as mentioned in [#9126](https://github.com/radareorg/radare2/issues/9126). As I was tinkering with this issue I found out that substitution of combination `[reg +/- offset]` by variable name depends on brackets being there, so if brackets are removed **before** above mentioned combination is substituted by name of the variable, then (as far as I understood) part of varsub function that iterates over ebp/esp variables doesn't consider this a variable and thus substitution doesn't occur. If you're referring to changing disassembly completely that is
```
> e asm.pseudo = false
0x0804843c      8d45f8         lea eax, var_8h
```
```
> e asm.pseudo = true
0x0804843c      8d45f8         eax = var_8h

```
Then for sure we can try this (as I understood from comments it was tried and it worked), but in my opinion it breaks expectation of how lea source operand should look.

Reply:
I don't think we should change how `lea` looks in the disassembly, because it may be confusing, as in general it appears with `[..]`